sound/soc/sof/ipc4.c

```diff
@@ -219,6 +219,15 @@ int sof_ipc4_init_msg_memory(struct snd_sof_dev *sdev)
 
 	/* TODO: get max_payload_size from firmware */
 	sdev->ipc->max_payload_size = SOF_IPC4_MSG_MAX_SIZE;
+	/*
+	 * In kernel a full IPC4 message is calculated to take the
+	 * struct sof_ipc4_msg size:
+	 * msg_total_size = sizeof(msg) + msg.data_size;
+	 *
+	 * The msg_total_size should be passed as msg_bytes for function expecting
+	 * a message size and underlaying code should take this into account.
+	 */
+	sdev->ipc->max_payload_size += sizeof(struct sof_ipc4_msg);
```

Comment:
I dont understand this @ujfalusi . The max payload for me refers to the MAX size of the mailbox. Why would we include the header and extension in it?

Reply:
This was exactly my concern.
I guess I can modify things that the msg_bytes is the optional data size which goes via the mailbox, but then the msg_bytes is kind of half backed as we still have the u64 header, which is a message by itself in some cases.